internal/v2/fakes2av2/fakes2av2_test.go

```diff
@@ -205,6 +206,112 @@ func TestSetUpSession(t *testing.T) {
 				},
 			},
 		},
+		{
+			description: "client side private key operation",
+			request: &s2av2pb.SessionReq {
+				LocalIdentity: &commonpbv1.Identity {
+					IdentityOneof: &commonpbv1.Identity_Hostname {
+						Hostname: "client_side_connection",
+					},
+				},
+				AuthenticationMechanisms: []*s2av2pb.AuthenticationMechanism {
+					{
+						// TODO(rmehta19): Populate Authentication Mechanism using tokenmanager.
+						MechanismOneof: &s2av2pb.AuthenticationMechanism_Token{"token"},
+					},
+				},
+				ReqOneof: &s2av2pb.SessionReq_OffloadPrivateKeyOperationReq {
+					&s2av2pb.OffloadPrivateKeyOperationReq {
+						Operation: s2av2pb.OffloadPrivateKeyOperationReq_SIGN,
+						SignatureAlgorithm: s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_RSA_PSS_RSAE_SHA256,
+						InBytes: []byte("Generate hash and sign this."),
+					},
+				},
+			},
+			expectedResponse: &s2av2pb.SessionResp {
+				Status: &s2av2pb.Status {
+					Code: 0,
+				},
+				RespOneof: &s2av2pb.SessionResp_OffloadPrivateKeyOperationResp {
+					&s2av2pb.OffloadPrivateKeyOperationResp {
+						OutBytes: []byte {
```

Comment:
I essentially populated OutBytes with nothing and after the first run of the test, I copy pasted the diff here, essentially assuming the output of fakes2av2 is correct. 

Looking back, I should probably be doing both of these things in this test file:
- creating the 256 bit Hash of the InBytes using the SHA256 function
- calling the sign method(implemented by the client/server cert/key pair) to sign with the client/server private key here

Reply:
Could we generate this data by signing using OpenSSL? I'm worried that the correctness of our test is based on values outputted by our test, if that makes sense. :)

Reply:
Yes that makes a lot of sense! I'll also add the documentation about what specific openssl command I use to generate the SHA256 hash.

Reply:
Actually, now that I am changing the fake S2A implementation of offloadPrivateKeyOperation to not compute the hash of the input (moving hash computation to client remote signer lib in next PR), is it ok if I use the crypto/sha256 package functions instead of openssl?